crates/sui-indexer/src/store/pg_indexer_store.rs

```diff
@@ -185,6 +188,38 @@ impl<T: R2D2Connection + 'static> PgIndexerStore<T> {
         .context("Failed reading latest epoch id from PostgresDB")
     }
 
+    /// Get the protocol version of the latest epoch stored in the database
+    pub fn get_latest_epoch_protocol_version(&self) -> Result<Option<i64>, IndexerError> {
+        read_only_blocking!(&self.blocking_cp, |conn| {
+            epochs::dsl::epochs
+                .select(max(epochs::protocol_version))
+                .first::<Option<i64>>(conn)
+        })
+        .context("Failed reading latest epoch protocol version from PostgresDB")
+    }
+
+    /// Get the latest protocol version stored in the protocol_configs table, not to be confused with the
+    /// function above.
+    pub fn get_latest_protocol_config_version(&self) -> Result<Option<i64>, IndexerError> {
+        read_only_blocking!(&self.blocking_cp, |conn| {
+            protocol_configs::dsl::protocol_configs
+                .select(max(protocol_configs::protocol_version))
+                .first::<Option<i64>>(conn)
+        })
+        .context("Failed reading latest protocol version from PostgresDB")
+    }
+
+    pub fn get_chain_identifier(&self) -> Result<Option<Vec<u8>>, IndexerError> {
+        read_only_blocking!(&self.blocking_cp, |conn| {
+            checkpoints::dsl::checkpoints
+                .select(checkpoints::checkpoint_digest)
+                .filter(checkpoints::sequence_number.eq(0))
```

Comment:
this always reads the `sequence_number.eq(0)`, what if we have pruned this or restore only the latest 30D data?

Reply:
Yes this is indeed an issue. We read from checkpoint 0 for chain identifier in other queries too. I think the right thing to do here is to have a separate `chain_identifier` singleton table that stores just the chain identifier and never gets pruned away. Wdyt? @gegaowp

Reply:
fixed in #18825